win: port obj_tx_locks_abort

ChandKV opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1118)
<!-- Reviewable:end -->

krzycz commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/test/obj_tx_locks_abort/TEST0.PS1, line 56 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQPiMT8HgbJlhiKe9n5:-KQPiMT9nRN3CRK1F-gR:b-tcfimh) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST0.PS1#L56)):*
> ```
> setup
> 
> expect_normal_exit $ENV:EXE_DIR\obj_tx_locks_abort$EXESUFFIX $DIR\testfile1
> ```

$Env:EXESUFFIX

---

*[src/test/obj_tx_locks_abort/TEST1.PS1, line 54 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQPj8kseXuWy-pjQiZ3:-KQPj8kseXuWy-pjQiZ4:bkh907f) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST1.PS1#L54)):*
> ```
> . ..\unittest\unittest.ps1
> 
> Remove-Item ENV:\PMEM_LOG_LEVEL
> ```

I'm not sure if it's equivalent, but:
`$Env:PMEM_LOG_LEVEL=""`
seems to be more convenient.
Also, is '\' really required here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1118)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/test/obj_tx_locks_abort/TEST1.PS1, line 54 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQPj8kseXuWy-pjQiZ3:-KQPktUEW_jsC9ARUo17:blahdru) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST1.PS1#L54)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> I'm not sure if it's equivalent, but:
> `$Env:PMEM_LOG_LEVEL=""`
> seems to be more convenient.
> Also, is '\' really required here?
> 
</details>
Or better:
`$Env:PMEM_LOG_LEVEL=$null`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1118)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/test/obj_tx_locks_abort/TEST2.PS1, line 45 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQQJu5X27OHqquNaSG9:-KQQJu5X27OHqquNaSGA:b-qkuak3) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST2.PS1#L45)):*
> ```
> $ENV:UNITTEST_NUM = "2"
> 
> # standard unit test setup
> ```

```
# XXX:  bash has a few calls to tools that we don't have on
# Windows (yet) that set PMEM_IS_PMEM and NON_PMEM_IS_PMEM based
# on their output
$ENV:PMEM_IS_PMEM = $true
$ENV:NON_PMEM_IS_PMEM = $true
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1118)*
<!-- Sent from Reviewable.io -->

ChandKV commented:
Review status: 5 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/obj_tx_locks_abort/TEST0.PS1, line 56 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQPiMT8HgbJlhiKe9n5:-KQT5bWCmMa0Zfnf0RTV:bt2vr1b) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST0.PS1#L56)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> $Env:EXESUFFIX
</details>
Fixed.

---

*[src/test/obj_tx_locks_abort/TEST1.PS1, line 54 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQPj8kseXuWy-pjQiZ3:-KQSfitkgscgqqTwF3Te:bdpqejp) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST1.PS1#L54)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> Or better:
> `$Env:PMEM_LOG_LEVEL=$null`
</details>
All three have the same effect. Yes "\" is required for Remove-Item. Changed to $null version.

---

*[src/test/obj_tx_locks_abort/TEST2.PS1, line 45 \[r1\]](https://reviewable.io:443/reviews/pmem/nvml/1118#-KQQJu5X27OHqquNaSG9:-KQT439SBtp_7Tsir8MD:b-vzqk93) ([raw file](https://github.com/pmem/nvml/blob/e3d46e801b9551fbd4d88a93ff525432813a9efe/src/test/obj_tx_locks_abort/TEST2.PS1#L45)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> ```
> # XXX:  bash has a few calls to tools that we don't have on
> # Windows (yet) that set PMEM_IS_PMEM and NON_PMEM_IS_PMEM based
> # on their output
> $ENV:PMEM_IS_PMEM = $true
> $ENV:NON_PMEM_IS_PMEM = $true
> ```
</details>
ah... my bad, thank you! Fixed.
I think I should figure out why the test doesn't fail in my environment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1118)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1118#-:-KQUp8ehWgObmb7hqnep:bnfp4nl)*
<!-- Sent from Reviewable.io -->
